src/main/java/mira/Storage.java

```diff
@@ -0,0 +1,102 @@
+package mira;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Handles saving and loading tasks from a file.
+ */
+public class Storage {
+    private final String filePath;
+
+    /**
+     * Constructs a Storage object with the specified file path.
+     *
+     * @param filePath The path to the file where tasks will be saved and loaded from.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Appends a single task to the storage file.
+     *
+     * @param task The task to be saved to the file.
+     * @throws IOException If an I/O error occurs while writing to the file.
+     */
+    public void saveTask(Task task) throws IOException {
+        try (FileWriter writer = new FileWriter(filePath, true)) { // Append mode
+            writer.write(task.toFileString() + System.lineSeparator());
+        }
+    }
+
+    /**
+     * Saves a list of tasks to the storage file, overwriting any existing content.
+     *
+     * @param tasks The list of tasks to be saved to the file.
+     * @throws IOException If an I/O error occurs while writing to the file.
+     */
+    public void saveTasks(ArrayList<Task> tasks) throws IOException {
+        FileWriter writer = new FileWriter(filePath);
+        for (Task task : tasks) {
+            writer.write(task.toFileString() + System.lineSeparator());
+        }
+        writer.close();
+    }
+
+    /**
+     * Loads the list of tasks from the file.
+     *
+     * @return A list of tasks loaded from the file.
+     * @throws IOException If an I/O error occurs during loading.
+     * @throws SecurityException If a permission error occurs during loading.
+     * @throws IllegalArgumentException If file content corrupts during loading.
+     * @throws MiraException If unsupported task type founds during loading.
```

Comment:
Small Nitpick: Is there a Grammatical Error? I think "founds" should be "found".